Storage/StorageTests/CoreData/MigrationTests.swift

```diff
@@ -1322,6 +1322,67 @@ final class MigrationTests: XCTestCase {
         // Check `filterKey` is correctly set.
         XCTAssertEqual(migratedProductSearchResults.value(forKey: "filterKey") as? String, "sku")
     }
+
+    func test_migrating_from_73_to_74_adds_Customer_and_CustomerSearchResult_entities() throws {
+        // Given
+        let sourceContainer = try startPersistentContainer("Model 73")
+        let sourceContext = sourceContainer.viewContext
+
+        try sourceContext.save()
+
+        // Confidence Check. These entities should not exist in Model 73
+        XCTAssertNil(NSEntityDescription.entity(forEntityName: "Customer", in: sourceContext))
+        XCTAssertNil(NSEntityDescription.entity(forEntityName: "CustomerSearchResult", in: sourceContext))
+
+        // When
+        let targetContainer = try migrate(sourceContainer, to: "Model 74")
+
+        // Then
+        let targetContext = targetContainer.viewContext
+
+        // This entity should exist in Model 74
+        XCTAssertNotNil(NSEntityDescription.entity(forEntityName: "Customer", in: targetContext))
+        XCTAssertNotNil(NSEntityDescription.entity(forEntityName: "CustomerSearchResult", in: targetContext))
+        XCTAssertEqual(try targetContext.count(entityName: "Customer"), 0)
+        XCTAssertEqual(try targetContext.count(entityName: "CustomerSearchResult"), 0)
+
+        // Insert a new Customer
+        let migratedCustomer = insertCustomer(to: targetContext)
```

Comment:
Not hugely important in this test, but just to comment for understanding:

Naming... strictly this isn't a `migratedCustomer`, imo. A migrated customer would be if the `Customer` entity already existed, and you added a property to it or changed it in some other way. We use it in other tests to differentiate between the pre-migration entity, and the post-migration entity. 

[For example, `product` and `migratedProduct` in this test.](https://github.com/woocommerce/woocommerce-ios/blob/de959a276be9c197d97855c482ab242a573bda9e/Storage/StorageTests/CoreData/MigrationTests.swift#L278)

The meaning of "migration" here is as follows: 

> Updating existing data to match changes to the schema.

e.g. if version 73 had:

```
Customer
 - name: String
 - dob: Date?
 - phone: String?
 - address: Address
```

and then you updated the entity in 74 to be:

```
Customer
 - name: String
 - dob: Date?
 - phone: String
 - address: Address
 - billingAddress: Address?
```

I.e. you made `phone` mandatory, and added an optional `billingAddress`...

You would write **a migration for the customer** which did the following (pseudocode):

```
for customer in allStoredCustomers {
    if customer.phone == nil {
        migratedCustomer.phone = ""
    }
    migratedCustomer.billingAddress = nil
    customer.replace(with: migratedCustomer)
}
```

In this case, your migration only adds `Customer` to the schema, so you're not migrating any existing customers or search results.

All that to say... you can drop the `migrated` from `migratedCustomer` and `migratedCustomerSearchResult` here.